var/spack/repos/builtin/packages/hipcc/package.py

```diff
@@ -13,24 +13,48 @@ class Hipcc(CMakePackage):
 
     homepage = "https://github.com/ROCm/hipcc"
     git = "https://github.com/ROCm/hipcc.git"
-    url = "https://github.com/ROCm/HIPCC/archive/refs/tags/rocm-6.0.2.tar.gz"
+
+    def url_for_version(self, version):
+        if version <= Version("6.0.2"):
+            url = "https://github.com/ROCm/HIPCC/archive/rocm-{0}.tar.gz"
+        else:
+            url = "https://github.com/ROCm/llvm-project/archive/rocm-{0}.tar.gz"
```

Comment:
For future reference, we recommend switching to Python f-strings (https://docs.python.org/3/tutorial/inputoutput.html) as we do the same in the core.